core/main/src/service/apps/delegated_launcher_handler.rs

```diff
@@ -105,9 +107,94 @@ pub struct AppManagerState {
     apps: Arc<RwLock<HashMap<String, App>>>,
     // Very useful for internal launcher where the intent might get untagged
     intents: Arc<RwLock<HashMap<String, NavigationIntent>>>,
+    // This is a map <app_id, app_title>
+    app_title: Arc<RwLock<HashMap<String, String>>>,
+    app_title_persist_path: String,
 }
 
 impl AppManagerState {
+    pub fn new(saved_dir: &str) -> Self {
+        let persist_path = Self::get_storage_path(saved_dir);
+        let persisted_app_titles = AppManagerState::load_persisted_app_titles(&persist_path);
+        AppManagerState {
+            apps: Arc::new(RwLock::new(HashMap::new())),
+            intents: Arc::new(RwLock::new(HashMap::new())),
+            app_title: Arc::new(RwLock::new(persisted_app_titles)),
+            app_title_persist_path: persist_path,
+        }
+    }
+    fn restore_app_info_from_storage(storage_path: &str) -> Result<Value, String> {
+        let file_path = format!("{}/{}", storage_path, APP_ID_TITLE_FILE_NAME);
+
+        let file = fs::OpenOptions::new()
+            .read(true)
+            .open(file_path)
+            .map_err(|error| format!("Failed to open AppInfo storage file: {}", error))?;
+
+        let data: Value = serde_json::from_reader(&file)
+            .map_err(|error| format!("Failed to read data from AppInfo storage file: {}", error))?;
+        Ok(data)
+    }
+    fn load_persisted_app_titles(storage_path: &str) -> HashMap<String, String> {
+        // let storage_path = Self::get_storage_path(saved_dir);
+        let stored_value_res = Self::restore_app_info_from_storage(storage_path);
+        if let Ok(stored_value) = stored_value_res {
+            let map_res: Result<HashMap<String, String>, _> = serde_json::from_value(stored_value);
+            map_res.unwrap_or_default()
+        } else {
+            HashMap::new()
+        }
+    }
+    fn get_storage_path(saved_dir: &str) -> String {
+        let mut path = std::path::Path::new(saved_dir).join("app_info/");
```

Comment:
Please check the usage of format!(" {} / {}" ) in file path.  Instead  use `PathBuf` or `Path` to build the file url. 
Avoid using explicit file path delimiter ("/"), let `join() `take care of that.